### PR TITLE
Create publish workflow

### PR DESCRIPTION
**Pull request by johnjeffers**

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+          aws-region: us-west-2
+
+      - name: get secret
+        run: |
+          while IFS=$'\t' read -r key value; do
+            echo "::add-mask::${value}"
+            echo "${key}=${value}" >> $GITHUB_ENV
+          done < <(aws secretsmanager get-secret-value \
+            --region us-west-2 \
+            --secret-id platform/npmjs \
```
Comment (mooreds) [inline .github/workflows/publish.yaml:55]: This looks like we have one secret for NPM? Does it make sense to have multiple secrets for each project?

File: .github/workflows/publish.yaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      - name: build the package
+        run: npm run build
+
+      - name: test publish to npmjs
+        if: inputs.command == 'test'
+        run: npm publish --dry-run
+
+      - name: publish to npmjs
+        if: inputs.command == 'publish'
+        run: npm publish
```
Comment (mooreds) [inline .github/workflows/publish.yaml:81]: Should you have the same `main` branch check here that you do in the `commit version changes`?

**Review by mooreds — COMMENTED** [review]
A few questions. Hard to test this, but the publish workflow looks materially the same as the https://github.com/FusionAuth/fusionauth-typescript-client/blob/main/.github/workflows/deploy.yaml one.

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (mooreds) [reply .github/workflows/publish.yaml:55]: I mean one secret per project, rather than a single secret across N projects.

Comment (johnjeffers) [reply .github/workflows/publish.yaml:55]: I would prefer not to, but I'm open to hearing arguments in favor

**Review by lyleschemmerling — DISMISSED** [review]

File: .github/workflows/publish.yaml (hunk 2 above)
Comment (johnjeffers) [reply .github/workflows/publish.yaml:81]: Yeah, probably. I'll add that.

**Pull request merged**